### PR TITLE
kubelet: don't check for mounted before update dsw PV size

### PR DESCRIPTION
**Pull request by huww98**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

We are still only calling NodeExpand after the volume is mounted.

avoid depending on ASW from dswp.findAndAddNewPods(). It is weird to determine desired state based on actual state.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

This is part of PR #122140 , which is already stuck for nearly one year. So here I split it to smaller PRs.

In short, to fix the issue mentioned in #122140 , I want to remove `outerVolumeSpecName` from ASW. This field had an out-of-sync issue before (https://github.com/kubernetes/kubernetes/issues/103143), and its fix seems already hacky. So I decide to just remove `outerVolumeSpecName` from ASW. This PR removes one of the usages.

This change has already been reviewed before at https://github.com/kubernetes/kubernetes/pull/122140#discussion_r1524931979 .

/cc @jsafrane

This PR only contains the commits that you have approved in #122140 . Could you please add your approval here too?

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go [hunk 1]
```diff
@@ -322,7 +308,7 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 		}
 
 		// Add volume to desired state of world
-		_, err = dswp.desiredStateOfWorld.AddPodToVolume(
+		uniqueVolumeName, err := dswp.desiredStateOfWorld.AddPodToVolume(
 			uniquePodName, pod, volumeSpec, podVolume.Name, volumeGIDValue, seLinuxContainerContexts[podVolume.Name])
```
Comment (jsafrane) [inline pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:312]: Now that I see it again, I can see `AddPodToVolume` returns an empty string on error and that empty string gets passed to `checkVolumeFSResize` below (even on error!). I think it is wrong.

Comment (huww98) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go:312]: Thanks! I think we should just skip `checkVolumeFSResize` on error. If `AddPodToVolume` failed, `getUniqueVolumeName`(removed in this PR) should return `!exists` anyway.

**Comment by jsafrane** [comment]
/lgtm
/approve
/test pull-kubernetes-e2e-gce
> NodeProblemDetector [NodeFeature:NodeProblemDetector] should run without error

**Pull request merged**